_posts/2020-08-20-codestarts-preview.adoc

```diff
@@ -0,0 +1,114 @@
+It is called Codestart. The idea is simple, you pick your extensions and they are able to provide example code to get you started with it :)
+
+This preview version is just the beginning and includes three example codestarts:
+
+- The legendary RestEasy example in Java, Kotlin and Scala, when you pick the RestEasy extension
+- The lovely and sweet template engine Qute in Java, when you pick the Qute extension
+- The tough and strong Quarkus Command Mode in Java, when you don’t pick any extension that provides an example
+
+== Wanna give it a try?
+
+We added a feature flag in the Quarkus Maven plugin to create your application:
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:create -DcodestartsEnabled`
+----
+
+=== A Command mode Quarkus project using Maven as build tool:
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:create -DcodestartsEnabled \
+-DprojectGroupId=org.acme -DprojectArtifactId=go-go-commando
+cd go-go-commando
+# ./mvnw.cmd on Windows
+./mvnw package
+java -jar target/go-go-commando-1.0-SNAPSHOT-runner.jar
+----
+
+===  Same as previous using Gradle as build tool:
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:create -DcodestartsEnabled \
+-DprojectGroupId=org.acme -DprojectArtifactId=go-go-commando-gradle \
+-DbuildTool=GRADLE
+cd go-go-commando-gradle
+# ./gradlew.bat on Window
+./gradlew build
+java -jar build/go-go-commando-gradle-1.0-SNAPSHOT-runner.jar
+----
+
+=== A Quarkus project in Kotlin with example for RESTEasy and Qute:
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:create -DcodestartsEnabled \
+-DprojectGroupId=org.acme -DprojectArtifactId=kotlin-with-rest-is-qute \
+-Dextensions=kotlin,resteasy,qute
+cd kotlin-with-rest-is-qute
+# ./mvnw.cmd on Windows
+./mvnw quarkus:dev
+----
+
+=== A blank Quarkus project in Scala with SmallRye Reactive Messaging Kafka
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:create -DcodestartsEnabled \
+-DprojectGroupId=org.acme -DprojectArtifactId=scala-kafka \
+-Dextensions=scala,quarkus-smallrye-reactive-messaging-kafka \
+-DskipExample
+cd scala-kafka
+# ./mvnw.cmd on Windows
+./mvnw quarkus:dev
+----
+
+NOTE: We added instruction to get you started (contextual to the extensions and build tool you've selected) in the new project `README.md` .
+
+== Start a blank Quarkus application with your favorites tools/extensions
+
+Codestarts also introduce a huge improvement 😅 and don’t laugh, it has been a legitimate demand from the community for a while now,
+using codestarts, you can generate a blank Quarkus app by adding: `-DskipExample`.
+
+== You might notice a few missing features with Codestarts
+
+1. The package name of the generated examples is now fixed to `org.acme`,
+we thought it wouldn’t add much value to make it customizable, but let us know if you find it’s a loss.
```

Comment:
Seriously? This is an issue IMHO. People obviously want a custom package and the first thing they will have to do is to rename the package.

Reply:
@gsmet this is all example code meant to be deleted, or used as example. We are not generating the user application, but showing example of what they can use in there app.. So they will have to pick what they want and do their cooking with it..

Reply:
@gsmet you can still set the group id for maven/gradle. For the package/class, the convention to use in example codestarts to avoid conflicts is:
`org.acme.[example-name].MyClass`

Also think that before you were able to set the class name (not just the package), now you have multiple examples, this feature is not really making sense anymore. IMO if this we didn't have an existing codegen with that feature (and I wouldn't have noted it), you wouldn't even have noticed :)

Reply:
The `classname` and `path` options were making sense because the current codegen is generating an application to start from. I would say this is required if we were implementing an openapi code generator for example, but this is not what we are trying to achieve here, those are just bits of code to get inspiration.

Reply:
lets open an issue on this if we feel strongly its needed (where we can discuss all the hoops we would need to jump through to support it - its more than just a prefix + folder rename). Shouldn't stop the blog to go out as this is the current situation.

Reply:
we already have an issue: https://github.com/quarkusio/quarkus/issues/11154